### ISSUE TITLE
"input" event should fire synchronously when input field value changes

### ISSUE THREAD

**Issue by robertknight** [bitovi/syn#120]
syn fires the "input" event asynchronously when typing into a field using `syn.type(...)`. According to [the spec](https://w3c.github.io/uievents/#event-type-input) and [MDN](https://developer.mozilla.org/en-US/docs/Web/Events/input) this event should be fired synchronously as soon as the `value` property of the input has been updated.

**Comment by mvorisek** [comment]
I am facing the same issue. Do you have any idea how to fix this lib?

**Comment by justinbmeyer** [comment]
it looks like `.schedule()` is called here around `input`: https://github.com/bitovi/syn/blob/master/src/key.js#L872


@mvorisek , if you are up for pairing on this problem, I can help you fix it.

**Comment by mvorisek** [comment]
@justinbmeyer thank you, I will create a repro where this issue corrupts data massively (vs. regular keyboard input) shortly.

**Comment by mvorisek** [comment]
@justinbmeyer here is the repro: https://github.com/atk4/ui/commit/8971618c2044796177313aa46b41bce123edbd45

steps to reproduce:

1. clone and checkout `repro_bitovi_syn_issue_120` branch
2. open `repro_bitovi_syn_issue_120/index.html` in your browser (tested in Firefox on my side)
3. observe text with missing letters written to the input field:
   ![image](https://user-images.githubusercontent.com/2228672/235470749-08ca82de-111f-46c7-84ed-d6d5b0379a00.png)

I would expect `Czech Republic` written instead. When `Czech Republic` is typed using keyboard, the text is written correctly.

I belive the problem is that the text is mutated due race-condition between vue and syn.js as the input event generated by syn.js is not synchronous.

PS: I have created https://github.com/bitovi/syn/pull/196 PR which fixes this issue and the repro. Is it all it was or do some test or anything else needs to be adjusted?

**Comment by justinbmeyer** [comment]
Yeah, a test would be great.  You can see them in here: https://github.com/bitovi/syn/blob/master/test/test.html

**Comment by mvorisek** [comment]
Can you please help me with the test and create a PR againts my PR branch? I cannot run the JS tests localy nor I am sure if I can write the test properly.

**Comment by justinbmeyer** [comment]
@mvorisek , yes. Would you be open to pairing with me on it? Thanks!

**Comment by mvorisek** [comment]
Thank you, I am very happy this will be fixed as the issue is quite major. Feel free to submit more changes to https://github.com/bitovi/syn/pull/196 or just reuse the fix in your own PR.

I also found some problem when using Syn.focus() which breaks Selenium and with older release the tests were passing... Investigating what events are fired by this lib and what events are fixed by regular mouse event.

Also in the git I see you were working on v1 release where you fixed a lot of things incl. CI, what is the status now? What about adding GH Actions CI and integrate the changes incrementally?

**Comment by justinbmeyer** [comment]
We paused development, mostly b/c we didn't know folks have been using the library much outside of Bitovi. What have you been using it for?

Knowing folks are using it helps motivate us to keep maintaining it.

**Comment by mvorisek** [comment]
This lib is used in https://github.com/minkphp/MinkSelenium2Driver/blob/master/src/Selenium2Driver.php to emulate mouse/keyboard events for Selenium testing, there are about 25k installs per day - https://packagist.org/packages/behat/mink-selenium2-driver/stats, in npm stats it is not visible, as the JS is bundled, and there are probably a lot of simillar uses.
